plugins/generic/pln/PLNPlugin.inc.php

```diff
@@ -175,12 +175,29 @@ function getStyleSheet() {
 	 * @param $settingName string
 	 */
 	function getSetting($journalId,$settingName) {
-
-		// if there isn't a journal_uuid, make one
-		if ($settingName == 'journal_uuid') {
-			$uuid = parent::getSetting($journalId, $settingName);
-			if ($uuid) return $uuid;
-			$this->updateSetting($journalId, $settingName, $this->newUUID());
+                $settingValue = parent::getSetting($journalId, $settingName);
+		if ($settingValue != null) return $settingValue;
+                        
+		// if a setting returns null, populate it with a default
+		switch ($settingName) {
```

Comment:
I think Chris already proposed a fix like this. I'd like to avoid it if at all possible -- having the settings updated when they're fetched is certain to have unintended consequences further down the road. If it's possible to reproduce the underlying problem, someone on the dev team can help to figure it out.